spec/models/user.spec.js

```diff
@@ -168,4 +168,93 @@ describe("User model", () => {
       });
     });
   });
+
+  it("returns two users if either match username or email", (done) => {
+    const user1 = new User({
+      username: "someone",
+      first_name: "some",
+      last_name: "one",
+      email: "someone@example.com",
+      password: "password",
+      friends: [],
+    });
+
+    const user2 = new User({
+      username: "someoneelse",
+      first_name: "other",
+      last_name: "person",
+      email: "someoneelse@example.com",
+      password: "password",
+      friends: [],
+    });
+
+    user1.save((err) => {
+      console.log(err);
+      expect(err).toBeNull();
+
+      user2.save((err) => {
+        expect(err).toBeNull();
+
+        User.find(
+          {
+            $or: [
+              { username: "someone" },
+              { email: "someoneelse@example.com" },
+            ],
+          },
+          (err, users) => {
+            expect(err).toBeNull();
+            expect(users.length).toBe(2);
+            done();
+          }
+        );
+      });
+    });
+  });
+
+  // it.only("returns an error if trying to save a user with a duplicate username or email", (done) => {
+  //   const user1 = new User({
+  //     username: "someone",
+  //     first_name: "some",
+  //     last_name: "one",
+  //     email: "someone@example.com",
+  //     password: "password",
+  //     friends: [],
+  //   });
+
+  //   const user2 = new User({
+  //     username: "someoneelse",
+  //     first_name: "other",
+  //     last_name: "person",
+  //     email: "someoneelse@example.com",
+  //     password: "password",
+  //     friends: [],
+  //   });
+
+  //   const user3 = new User({
+  //     username: "someoneelse",
+  //     first_name: "some",
+  //     last_name: "one",
+  //     email: "someone@example.com",
+  //     password: "password",
+  //     friends: [],
+  //   });
+
+  //   user1.save((err) => {
+  //     // console.log(err);
+  //     expect(err).toBeNull();
+
+  //     user2.save((err) => {
+  //       expect(err).toBeNull();
+
+  //       user3
+  //         .save(() => {})
+  //         .catch((err) => {
+  //           console.log(err);
+  //           expect(err.keyPattern.username).toBe(1);
+  //           done();
+  //         });
+  //     });
+  //   });
+  // });
```

Comment:
Is this supposed to be commented out, this seems exactly what we need to test, no?